app/assets/javascripts/test/tracing-view-screenshot-tests/dataset_rendering.screenshot.js

```diff
@@ -0,0 +1,124 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+const test = require("ava");
+const puppeteer = require("puppeteer");
+const pixelmatch = require("pixelmatch");
+const { PNG } = require("pngjs");
+const fs = require("fs");
+const path = require("path");
+const { screenshotDataset, DEV_AUTH_TOKEN } = require("./dataset_rendering_helpers");
+
+process.on("unhandledRejection", (err, promise) => {
+  console.error("Unhandled rejection (promise: ", promise, ", reason: ", err, ").");
+});
+
+const BASE_PATH = path.join(__dirname, "../screenshots");
+
+let BRANCH = "master";
+if (!process.env.BRANCH) {
+  console.warn(
+    "[Warning] No branch specified, assuming branch master. If you want to specify a branch, prepend BRANCH=<branchname> to the command.",
+  );
+} else {
+  BRANCH = process.env.BRANCH;
+  console.log(`[Info] Executing tests on branch ${BRANCH}.`);
+}
+
+function compareScreenshot(screenshotBuffer, name, width, height) {
```

Comment:
This function is a bit hard to read in my opinion and could probably be improved by wrapping the png library with promise-based functions. But it's probably fine either way, since it's "just" test code :)

Reply:
Yes, these node function are pretty ugly to work with imo if you're used to async/await and promises. I've reworked that part and it should be much better now :)